arch/arm/src/Makefile

```diff
@@ -18,7 +18,7 @@
 #
 ############################################################################
 
-include $(TOPDIR)/Make.defs
+include $(TOPDIR)/apps-or-nuttx-Make.defs
```

Comment:
let's simple add:
```
-include $(APPDIR)/Make.defs
```
with the comment?

Reply:
So the apps dir will be required even for a kernel only build of NuttX?

Reply:
but then it won't have included `nuttx/Make.defs`.

The logic is "if there is `apps/Make.defs`, include it (which in turn includes `nuttx/Make.defs`, otherwise include `nuttx/Make.defs` directly"

Reply:
Yes, you right

Reply:
> Yes, you right. But can we do the trick inside Config.mk?

I tried, however the arch Makefiles include directly `nuttx/Make.defs` -- the board `Make.defs`, (which will eventually include `Config.mk`) and then `apps/Make.defs` would be included after we have already included `nuttx/Make.defs` so it won't work :(

Maybe a slightly cleaner solution is link the board `Make.defs` something else, like `nuttx/Board.defs` and then have this new file `apps-or-nuttx-Make.defs` be called `nuttx/Make.defs`, but this is essentially the same solution with better file naming.